src/pages/cli/project/troubleshooting.mdx

```diff
@@ -0,0 +1,245 @@
+|amplify-cfn-templates/ |folder| CloudFormation templates generated by Amplify for each resource category |
+|amplify-meta.json| file | Used by CLI core and the plugins to log internal information and to communicate with each other |
+|backend-config.json | file | Configuration about how your project's backend connects to AWS resources | 
+|hooks/ | folder | User provided Amplify hooks scripts |
+|root-cloudformation-stack.json | file | The parent CloudFormation template for your application's nested stack |
+
+
+** The following sections contain examples of different types of "amplify push" failures and suggested resolutions **
+
+### Push failures from unmanaged changes to cloud resources - Drift 
+<Callout warning>
+
+**Warning:** Once an application is deployed to the cloud using Amplify CLI the cloud resources like DynamoDB tables, Cognito Roles, IAM policies must never be updated outside of Amplify CLI or Amplify Studio.
+
+</Callout>
+
+When a resource is deployed through Amplify is updated outside of Amplify this is called an unmanaged change. 
+The Amplify CLI cannot track such changes since they are applied outside of the Cloudformation generated by Amplify to provision those resources. 
+This difference between the expected configuration of the resource in the Cloudformation vs the actual configuration of the service is called a “CloudFormation drift”. 
+Currently there is no automated way to reverse a CloudFormation drift. Each resource must be inspected through the AWS console for drift in it's CloudFormation stack and the drift has to be resolved manually.
+
+Assume, using Amplify CLI you have created a GraphQL schema with multiple models, indexes and resolvers and deployed it to the cloud. 
+On deployment of this application, the Amplify generated CloudFormation template provisions AppSync API models, resolvers, multiple DynamoDB tables and GSIs. 
+
+#### Scenario 1: "amplify push" fails after manually updating DynamoDB tables and GSIs from the console.
+Assume you have an application with a GraphQL schema deployed to the cloud. 
+Amplify creates DynamoDB tables for all [@model]((/cli/graphql/data-modeling/)) annotations and GSIs for all [@index]((/cli/graphql/data-modeling/)) fields in the GraphQL schema. 
+Now, if you login into the DynamoDB console and delete any of these GSIs, this introduces a drift in the application. 
+This may cause future changes to the GraphQL schema to fail deployment. 
+As an example, if you change the names of some @models and @index fields in your GraphQL schema and perform `amplify push`with`—allow-destructive-graphql-schema-updates`flag enabled, Amplify will first remove all DynamoDB tables corresponding to the original model names. 
+However if any of the GSIs on these tables were already deleted manually from the console then the deployment will fail. This is because Amplify’s CloudFormation is attempting to update the state of a resource (the GSI) which doesn’t exist. 
```

Comment:
```suggestion
However if any of the GSIs on these tables were already deleted manually from the console then the deployment will fail. This is because Amplify’s CloudFormation stack is attempting to update the state of a resource (the GSI) which doesn’t exist. 
```